src/algorithm/minkowski/gjk/mod.rs

```diff
@@ -292,35 +398,168 @@ where
-                        match *strategy {
-                            CollisionStrategy::CollisionOnly => {
-                                return Some(Contact::new(CollisionStrategy::CollisionOnly));
+                    match *strategy {
+                        CollisionOnly => {
+                            return Some(contact);
+                        }
+                        FullResolution => contacts.push(contact),
+                    }
+                }
+            }
+        }
+
+        // CollisionOnly handling will have returned already if there was a contact, so this
+        // scenario will only happen when we have a contact in FullResolution mode
+        if contacts.len() > 0 {
+            // penetration depth defaults to 0., and can't be nan from EPA,
+            // so unwrapping is safe
+            contacts.into_iter().max_by(|l, r| {
+                l.penetration_depth
+                    .partial_cmp(&r.penetration_depth)
+                    .unwrap()
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Compute the distance between the given shapes.
+    ///
+    /// ## Parameters:
+    ///
+    /// - `left`: left shape
+    /// - `left_transform`: model-to-world-transform for the left shape
+    /// - `right`: right shape,
+    /// - `right_transform`: model-to-world-transform for the right shape
+    ///
+    /// ## Returns:
+    ///
+    /// Will optionally return the smallest distance between the objects. Will return None, if the
+    /// objects are colliding.
+    pub fn distance_complex<P, PL, PR, TL, TR>(
+        &self,
+        left: &[(PL, TL)],
+        left_transform: &TL,
+        right: &[(PR, TR)],
+        right_transform: &TR,
+    ) -> Option<P::Scalar>
+    where
+        P: EuclideanSpace,
+        P::Scalar: BaseFloat,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace,
+        PL: SupportFunction<Point = P>,
+        PR: SupportFunction<Point = P>,
+        TL: Transform<P>,
+        TR: Transform<P>,
+        SP: SimplexProcessor<Point = P>,
+    {
+        let mut min_distance = None;
+        for &(ref left_primitive, ref left_local_transform) in left.iter() {
+            let left_transform = left_transform.concat(left_local_transform);
+            for &(ref right_primitive, ref right_local_transform) in right.iter() {
+                let right_transform = right_transform.concat(right_local_transform);
+                match self.distance(
+                    left_primitive,
+                    &left_transform,
+                    right_primitive,
+                    &right_transform,
+                ) {
+                    None => return None, // colliding,
+                    Some(distance) => {
+                        min_distance = match min_distance {
+                            None => Some(distance),
+                            Some(min_distance) => {
+                                if distance < min_distance {
```

Comment:
`Some(cmp::min(...))` ?

Reply:
float does not form a complete ordering, and therefore have no Ord implementation, so can't use std::cmp unfortunately.